docs/custom-gateway.md

```diff
@@ -143,13 +144,46 @@ For detailed implementation, check out [Calendar gRPC gateway](https://github.co
 * To write gateway gRPC server using other languages than go, generate server interfaces using protoc.
 Follow protobuf tutorials []()https://developers.google.com/protocol-buffers/docs/tutorials
 
+## HTTP Gateway
+A gRPC gateway has 3 components, 
+1.  Gateway Processor Server - your implementation of HTTP streaming server, either generates events or listens to 
+external events and streams them back to gateway-processor-client. User code must accept POST requests on `/start` and `/stop`
+endpoints.
+
+2. Gateway Processor Client - sends configuration to gateway-processor-server on either `/start` endpoint for
+a new configuration or `/stop` endpoint to stop a configuration. Processor client itself has a HTTP server 
+running internally listening for events from gateway-processor-server.
+
+3. Gateway Transformer: transforms incoming events into cloudevents specification compliant events 
+   and dispatches them to interested sensors. 
+
+
+### Architecture
+![](http-gateway.png)
+
+List of environment variables available to user code
+
+|  Field               |  Description |
+|----------------------|--------------|
+| GATEWAY_PROCESSOR_SERVER_HTTP_PORT     | Gateway processor server HTTP server port |
+|  GATEWAY_PROCESSOR_CLIENT_HTTP_PORT | Gateway processor client HTTP server port  |
+|  GATEWAY_HTTP_CONFIG_START            | REST endpoint to post new configuration |
+|  GATEWAY_HTTP_CONFIG_STOP             | REST endpoint to make configuration to stop |
+|  GATEWAY_HTTP_CONFIG_EVENT             | REST endpoint to send events to |
+
+
+For detailed implementation, check out [Calendar HTTP gateway](https://github.com/argoproj/argo-events/tree/eventing/gateways/rest/calendar)
+
+###### Advantage of writing a gRPC or http gateway is that you can use other languages for implementation of gateway processor server.   
```

Comment:
isn't this an advantage for all types of gateways? as long as the gateway processor server has a library to interface with the gateway client.. be that NATS, Kafka, HTTP. you just need a supported protocol + infra (if needed in case of streams)